src/System.Management.Automation/engine/Subsystem/CommandPrediction/CommandPrediction.cs

```diff
@@ -27,15 +27,21 @@ public sealed class PredictionResult
         /// </summary>
         public string Name { get; }
 
+        /// <summary>
+        /// Gets the mini-session id that represents a specific invocation to the <see cref="ICommandPredictor.GetSuggestion"/> API of the predictor.
+        /// </summary>
+        public uint Session { get; }
```

Comment:
Why uint? Maybe Guid to be globally independent?

Reply:
The session is always under a context, for example, a period of time when a predictor module is loaded and used, a Runspace instance id where the module is loaded. That context can be uniquely identified, so I think there is no need for the mini-session to be globally unique.

I use `uint` only because it provides a large space and easy to increment to get a new mini-session id. Azure PowerShell team is using this prediction API for the [Az Predictor](https://techcommunity.microsoft.com/t5/azure-tools/announcing-az-predictor/ba-p/1873104). I'm waiting on the their feedback on this.

Reply:
The Az.Predictor team has confirmed that `uint` works fine, becuase they already have other context information that is uniquely identifiable. /cc @kceiw